lnwallet/chanfunding/coin_select.go

```diff
@@ -134,62 +185,31 @@ func CoinSelectSubtractFees(feeRate chainfee.SatPerKWeight, amt,
 		return nil, 0, 0, err
 	}
 
-	var weightEstimate input.TxWeightEstimator
-	for _, utxo := range selectedUtxos {
-		switch {
-
-		case txscript.IsPayToWitnessPubKeyHash(utxo.PkScript):
-			weightEstimate.AddP2WKHInput()
-
-		case txscript.IsPayToScriptHash(utxo.PkScript):
-			weightEstimate.AddNestedP2WKHInput()
-
-		default:
-			return nil, 0, 0, fmt.Errorf("unsupported address "+
-				"type: %x", utxo.PkScript)
-		}
-	}
-
-	// Channel funding multisig output is P2WSH.
-	weightEstimate.AddP2WSHOutput()
-
-	// At this point we've got two possibilities, either create a
-	// change output, or not. We'll first try without creating a
-	// change output.
-	//
-	// Estimate the fee required for a transaction without a change
+	// Obtain fee estimates both with and without using a change
 	// output.
-	totalWeight := int64(weightEstimate.Weight())
-	requiredFee := feeRate.FeeForWeight(totalWeight)
+	requiredFeeNoChange, requiredFeeWithChange, err := calculateFees(
+		selectedUtxos, feeRate)
+	if err != nil {
+		return nil, 0, 0, err
+	}
 
 	// For a transaction without a change output, we'll let everything go
 	// to our multi-sig output after subtracting fees.
-	outputAmt := totalSat - requiredFee
+	outputAmt := totalSat - requiredFeeNoChange
 	changeAmt := btcutil.Amount(0)
 
 	// If the the output is too small after subtracting the fee, the coin
 	// selection cannot be performed with an amount this small.
 	if outputAmt <= dustLimit {
 		return nil, 0, 0, fmt.Errorf("output amount(%v) after "+
 			"subtracting fees(%v) below dust limit(%v)", outputAmt,
-			requiredFee, dustLimit)
+			requiredFeeNoChange, dustLimit)
 	}
 
-	// We were able to create a transaction with no change from the
-	// selected inputs. We'll remember the resulting values for
-	// now, while we try to add a change output. Assume that change output
-	// is a P2WKH output.
-	weightEstimate.AddP2WKHOutput()
-
-	// Now that we have added the change output, redo the fee
-	// estimate.
-	totalWeight = int64(weightEstimate.Weight())
-	requiredFee = feeRate.FeeForWeight(totalWeight)
-
 	// For a transaction with a change output, everything we don't spend
 	// will go to change.
+	newOutput := amt - requiredFeeWithChange
```

Comment:
why move the line?

Reply:
Just for consistency, because above and below "output" comes before "change", so scanning the code is a little bit easier.